packages/expo-image-picker/ios/ImagePickerModule.swift

```diff
@@ -89,6 +90,16 @@ public class ImagePickerModule: Module, OnMediaPickingResultHandler {
     }
 
     let imagePickerDelegate = ImagePickerHandler(onMediaPickingResultHandler: self, hideStatusBarWhenPresented: options.allowsEditing)
+
+    let pickingContext = PickingContext(promise: promise,
+                                        options: options,
+                                        imagePickerHandler: imagePickerDelegate)
+
+    if #available(iOS 14, *), options.allowsMultipleSelection && !options.allowsEditing && sourceType != .camera {
+      self.launchMultiSelectPicker(pickingContext: pickingContext, currentViewController: currentViewController)
+      return
+    }
+
```

Comment:
Ok, I see that this function is growing in logic and I kind of dislike it 😅 
I suggest making this function a very top level and push the logic to other extracted functions 🤓 
I would like to ask you to provide something like this:
```swift
return switch (options) {
.case shouldLaunchMultiselectionPicker -> lauchMultiselectorPicker(...some options)
.case shouldLaunchLegacyPicker -> launchLegacyPicker(...some options)
.case shouldLaunchCamera -> launchCamera(...some options)
}
```

I guess some of `launch*` methods might share common logic:
```swift
private func launchLegacyPicker() {
  let picker = UIImagePickerController(options) // use can use https://docs.swift.org/swift-book/LanguageGuide/Extensions.html
  
  // add some specific logic
  picker.sourceType = .photoLibrary
  // etc

  launchPicker(picker)
}

private func launchCamera() {
  let picker = UIImagePickerController(options) // use can use https://docs.swift.org/swift-book/LanguageGuide/Extensions.html
  
  // add some specific logic
  picker.sourceType = .camera
  // etc

  launchPicker(picker)
}
```
When I look at this suggestion of mine I come to the conclusion that `.camera` and `.legacyPicker` are almost the same 😅

Reply:
Cleaned it up a bit, but left camera and legacy picker together for now.